File: LeetCode/1051高度检查器.cpp
int cmp(const void *a, const void *b) {
    return *(int *)a - *(int *)b;
}

int heightChecker(int* heights, int heightsSize){
    int cnt = 0;
    int *nums = (int *)malloc(sizeof(int) * heightsSize);
    for (int i = 0; i < heightsSize; i++) {
        nums[i] = heights[i];
    }
    qsort(nums, heightsSize, sizeof(int), cmp);
    for (int i = 0; i < heightsSize; i++) {
        if (nums[i] != heights[i]) cnt++;
    }
    return cnt;
}

File: LeetCode/1051高度检查器.go
func heightChecker(heights []int) int {
    var backup = make([]int, len(heights))
    copy(backup, heights)
    // sort.Sort(sort.IntSlice(backup))
    sort.Ints(backup)
    cnt := 0
    for i := 0; i < len(backup); i++ {
        if heights[i] != backup[i] {
            cnt++
        }
    }
    return cnt
}